modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -0,0 +1,190 @@
+                               throws BadRequestException, NotFoundException, CharonException {
+
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Users");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getUser(id, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        Assert.assertEquals(obj.getString("id"), id);
+    }
+
+    @DataProvider(name = "dataForGetThrowingExceptions")
+    public Object[][] dataToGetThrowingExceptions() {
+
+        return new Object[][]{
+                {"1234", "userName", null}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetThrowingExceptions")
+    public void testGetThrowingExceptions(String id, String attributes,
+                     String excludeAttributes) throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Users");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getUser(id, requiredAttributes)).thenReturn(null);
+
+        SCIMResponse outputScimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+
+        Assert.assertNull(outputScimResponse);
+    }
+
+    @DataProvider(name = "dataForListWithGetInt")
+    public Object[][] dataToGetListInt() throws CharonException, BadRequestException, InternalErrorException {
+
+        return new Object[][]{
+                {null, 1, 2, null, null, "PRIMARY", "emails", null, 200}
```

Comment:
Can we use existing constants for this `"PRIMARY"` word?

Reply:
Fixed with 39f1812bad7f3cc5344e1b3327f8f54e5e995532